app/upgrades/v16/utils_test.go

```diff
@@ -0,0 +1,266 @@
+	wevmosContract     common.Address
+}
+
+const (
+	AEVMOS = "aevmos"
+	XMPL   = "xmpl"
+
+	testAccount   = 0
+	erc20Deployer = testAccount + 1
+)
+
+// mintAmount is the amount of tokens to be minted for a non-native ERC20 contract.
+var mintAmount = big.NewInt(5e18)
+
+// SetupConvertERC20CoinsTest sets up a test suite to test the conversion of ERC20 coins to native coins.
+//
+// It sets up a basic integration test suite, with two accounts, that contain balances in a native non-Evmos coin.
+// This coin is registered as an ERC20 token pair, and a portion of the initial balance is converted to the ERC20
+// representation.
+func SetupConvertERC20CoinsTest(t *testing.T) (ConvertERC20CoinsTestSuite, error) {
+	kr := testkeyring.New(2)
+	fundedBalances := []banktypes.Balance{
+		{
+			Address: kr.GetAccAddr(testAccount).String(),
+			Coins: sdk.NewCoins(
+				sdk.NewCoin(AEVMOS, network.PrefundedAccountInitialBalance),
+				sdk.NewInt64Coin(XMPL, 300),
+			),
+		},
+		{
+			Address: kr.GetAccAddr(erc20Deployer).String(),
+			Coins: sdk.NewCoins(
+				sdk.NewCoin(AEVMOS, network.PrefundedAccountInitialBalance),
+				sdk.NewInt64Coin(XMPL, 200),
+			),
+		},
+	}
+
+	nw := network.NewUnitTestNetwork(
+		// TODO: we'll need to merge main into the feature branch to update this
+		network.WithBalances(fundedBalances...),
+	)
+	handler := grpc.NewIntegrationHandler(nw)
+	txFactory := testfactory.New(nw, handler)
+
+	govParamsRes, err := handler.GetGovParams("voting")
+	require.NoError(t, err, "failed to get gov params")
+	govParams := govParamsRes.GetParams()
+	govParams.MinDeposit = sdk.Coins{}
+
+	err = nw.UpdateGovParams(*govParams)
+	require.NoError(t, err, "failed to update gov params")
+
+	// Register the coins
+	XMPLMetadata := banktypes.Metadata{
+		Name:        XMPL,
+		Symbol:      XMPL,
+		Description: "Example coin",
+		DenomUnits: []*banktypes.DenomUnit{
+			{Denom: XMPL, Exponent: 0},
+			{Denom: "u" + XMPL, Exponent: 6},
+		},
+		Base:    XMPL,
+		Display: XMPL,
+	}
+
+	tokenPair, err := nw.App.Erc20Keeper.RegisterCoin(nw.GetContext(), XMPLMetadata)
+	require.NoError(t, err, "failed to register coin")
+
+	err = nw.NextBlock()
+	require.NoError(t, err, "failed to execute block")
+
+	// Call the token pair contract to check the balance
+	balance, err := GetERC20Balance(txFactory, kr.GetAddr(testAccount), tokenPair.GetERC20Contract())
+	require.NoError(t, err, "failed to query ERC20 balance")
+	require.Equal(t, common.Big0.Int64(), balance.Int64(), "expected different balance initially")
+
+	err = nw.NextBlock()
+	require.NoError(t, err, "failed to execute block")
+
+	// Convert the native coins to the ERC20 representation
+	senderIdx := 0
+	// FIXME: this method is removed on the feature branch -> use custom genesis instead?
+	msgConvert := &erc20types.MsgConvertCoin{
+		Coin:     sdk.NewCoin(XMPL, sdk.NewInt(100)),
+		Receiver: kr.GetAddr(senderIdx).String(),
+		Sender:   kr.GetAccAddr(senderIdx).String(),
+	}
```

Comment:
This flow won't work because the `MsgConvertCoin` is removed. Will have to look for another approach, probably using the custom `GenesisState` for the relevant modules.